draft-packages/button/KaizenDraft/Button/components/GenericButton.tsx

```diff
@@ -51,14 +51,39 @@ type Props = ButtonProps & {
   iconButton?: boolean
 } & AdditionalContentProps
 
-const GenericButton: React.FunctionComponent<Props> = props => (
-  <span
-    className={classNames(styles.container, {
-      [styles.fullWidth]: props.fullWidth,
-    })}
-  >
-    {props.href && !props.disabled ? renderLink(props) : renderButton(props)}
-  </span>
+export type ButtonRef = { focus: () => void }
+
+// We're treating custom props as anything that is kebab cased.
+// This is so we can support properties like aria-* or data-*
+const getCustomProps = (props: object) => {
+  const keys = Object.keys(props).filter(k => k.indexOf("-") !== -1)
+  return keys.reduce((acc, val) => {
+    acc[val] = props[val]
+    return acc
+  }, {})
+}
```

Comment:
Is this for kebab attributes other than `aria-*`and `data-*`? I would have thought native attributes like that would be supported by the `HTMLElement` types but not sure if this is for something else?

Reply:
So it's not quite related to the `HTMLElement` type. This is just to funnel any kebab props, including `aria-` and `data-` down to the actual `button` element (although I'm not sure if there are any other valid kebab attributes that we need to allow).